lib/logstash/inputs/tcp.rb

```diff
@@ -153,8 +159,31 @@ def server_connection_thread(output_queue, socket)
 
   def handle_socket(socket, client_address, client_port, output_queue, codec)
     peer = "#{client_address}:#{client_port}"
+    first_read = true
     while !stop?
-      codec.decode(read(socket)) do |event|
+      tbuf = read(socket)
+      if @proxy_protocol == 1 && first_read
```

Comment:
I figured the overhead of a single variable comparison would be dwarfed by the time taken for a read.